### DISCUSSION TITLE
Seeking Assistance for FeTS Functionality on Ubuntu 20

### DISCUSSION THREAD

**Discussion by Alhonainy21** [FeTS-AI/Front-End#70] [Q&A]
Hello,

I'm attempting to use FeTS on Ubuntu 20, utilizing my machine's GPU capabilities. However, I've come across information indicating that FeTS may not be functioning optimally on this OS. If anyone has successfully run FeTS on Ubuntu 20, could you please provide some guidance?

Thank you!

**Comment by sarthakpati** [comment]
Thanks for your interest in the FeTS Tool! Please see the following links:

- Guide for Ubuntu 20.04 users: https://fets-ai.github.io/FL-Pre/setup#note-for-ubuntu-2004-users
- FeTS Tool is now available as a Docker: https://github.com/FeTS-AI/Front-End/releases/tag/1.0.1

**Reply by sarthakpati** [reply]
You should be able to pass the `-d` parameter (and others) directly to the image:

```bash
docker run --rm ghcr.io/fets-ai/front-end:1.0.1 -d /path/to/your/datasets
```

**Reply by Alhonainy21** [reply]
Is the shape of dataset shown below ready to pass?
<img width="281" alt="Screenshot 2023-07-24 at 7 52 06 PM" src="https://github.com/FeTS-AI/Front-End/assets/92266474/9f77f4c0-bf58-4f39-b096-1489be1be825">

**Reply by Alhonainy21** [reply]
I passed a directory with a previous shape, however i got this error:
<img width="906" alt="Screenshot 2023-07-24 at 9 01 47 PM" src="https://github.com/FeTS-AI/Front-End/assets/92266474/91bc8e65-9e3c-4d13-af9c-68e44104d2fe">

could you please suggest the solution?

**Reply by sarthakpati** [reply]
Hi,

Please refer to [the documentation](https://fets-ai.github.io/FL-Pre/process_data#pre-processing) on how previously preprocessed data should be structured. `FeTS_CLI_Segment` [expects this directory structure](https://fets-ai.github.io/FL-Pre/runningApplication):

```
/path/to/output
│   │
│   └───DataForFeTS # this is to be passed for inference/training
│   │   │
│   │   └───Patient_001 # this is constructed from the ${PatientID} header of CSV
│   │   │   │ Patient_001_brain_t1.nii.gz
│   │   │   │ Patient_001_brain_t1ce.nii.gz
│   │   │   │ Patient_001_brain_t2.nii.gz
│   │   │   │ Patient_001_brain_flair.nii.gz
```

**Reply by Alhonainy21** [reply]
Hello,
Thank you for your attention to this matter.

From my understanding of the provided documentation, there are two steps necessary to achieve the aforementioned outcome:

First, a CSV file needs to be created. This has already been accomplished.
Secondly, the created CSV file must be passed through either the 'PrepareDataset' or 'PrepareDataset.py' code.
Upon implementing the second step, I encountered two distinct errors. Although I believe these errors are interrelated, I am unsure of how to resolve them. Please find the error screenshots attached for your perusal. Any guidance on how to resolve these issues would be highly appreciated.

<img width="1166" alt="Screenshot 2023-07-25 at 4 27 56 PM" src="https://github.com/FeTS-AI/Front-End/assets/92266474/5bbc3945-e7a1-4903-81c6-0a53cf16051d">

<img width="1170" alt="Screenshot 2023-07-25 at 4 28 31 PM" src="https://github.com/FeTS-AI/Front-End/assets/92266474/2716cb65-410f-4fb4-a34f-09134974dcda">

**Reply by sarthakpati** [reply]
Hmm, I am not sure how this is happening, since the Qt libraries should be getting put into the path. I am going to investigate this and let you know what I find, thanks.

**Comment by Alhonainy21** [comment]
I really appreciate your help!

This screenshot may help to understand why the error happened.

<img width="1170" alt="Screenshot 2023-07-25 at 5 07 38 PM" src="https://github.com/FeTS-AI/Front-End/assets/92266474/e2215170-f07c-4cb1-904e-0b6d2566352e">

**Reply by sarthakpati** [reply]
Are you using the docker image?

**Comment by Alhonainy21** [comment]
To use a Docker image, I understood that I must pass the data after preprocessing. If I don't pass the directory to the Docker image, it shows as 'exited' immediately. So, I am performing the preprocessing without using the Docker image.

I just solved the previous error by putting export LD_LIBRARY_PATH=${fets_root_dir}/lib:$LD_LIBRARY_PATH in ".bashrc" file.

**Reply by sarthakpati** [reply]
Great, but you can do the preprocessing using the Docker image as well.

**Comment by Alhonainy21** [comment]
Thank you so much!!
I really like the active discussions on this platform that make things easier for all users.

**Reply by sarthakpati** [reply]
Glad you're finding this useful!

**Comment by Alhonainy21** [comment]
I'm excited to use this platform for registration and segmentation. 
I hope the process will be smooth, and I will share my experiences here.
Thanks!

**Reply by sarthakpati** [reply]
Awesome, thanks
